### PR TITLE
fix(TDP-3475): fetch dataset list when editing a lookup step

### PR DESCRIPTION
**Pull request by charlesnguyen**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-3475

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [x] Functional tests have been performed

**Please check the browsers you've tested on**
- [x] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [ ] No, and no need to (backend changes only)

**(Optional) What is the current behavior?**
(Additional information to the Jira)

**(Optional) What is the new behavior?**
(Additional information to the Jira)

**(Optional) Other information**:

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/940/) [:star:](https://ci-tdp.datapwn.com/job/tdp_pull-request_master/940/)

File: dataprep-webapp/src/app/services/lookup/lookup-service.js [hunk 1]
```diff
@@ -102,28 +102,31 @@ export default class LookupService {
...
-				this.StateService.setGridSelection([selectedColumn]);
-
-				// lookup already loaded
-				if (this.state.playground.lookup.dataset === lookupAction &&
-					this.state.playground.lookup.step === step) {
-					return;
-				}
-
-				// load content
+		return this._getDatasets()
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/lookup/lookup-service.js]: Could you chnage the unit tests please?

Comment (charlesnguyen) [inline dataprep-webapp/src/app/services/lookup/lookup-service.js]: done

File: dataprep-webapp/src/app/services/lookup/lookup-service.js [hunk 2]
```diff
@@ -102,28 +102,31 @@ export default class LookupService {
...
+			.then(() => {
 				return this._getActions(this.state.playground.dataset.id)
-					.then(() => this.DatasetRestService.getContent(lookupId, true))
-					.then((lookupDsContent) => {
-						this._initLookupState(lookupAction, lookupDsContent, step);
+					.then((actions) => {
+						const lookupId = step.actionParameters.parameters.lookup_ds_id;
+						const lookupAction = _.find(actions, (action) => {
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/lookup/lookup-service.js]: Avoid lodash
Use `const lookupAction = actions.find(action => this._getDsId(action) === lookupId);` instead

File: dataprep-webapp/src/app/services/lookup/lookup-service.js [hunk 3]
```diff
@@ -102,28 +102,31 @@ export default class LookupService {
...
+							return this._getDsId(action) === lookupId;
+						});
+
+						// change column selection to focus on step target
+						const selectedColumn = _.find(this.state.playground.data.metadata.columns, { id: step.actionParameters.parameters.column_id });
```
Comment (frassinier) [inline dataprep-webapp/src/app/services/lookup/lookup-service.js]: Avoid lodash
Use `const selectedColumn = this.state.playground.data.metadata.columns.find(column => column.id === step.actionParameters.parameters.column_id );` instead

File: dataprep-webapp/src/app/services/lookup/lookup-service.js (hunk 1 above)
Comment (frassinier) [inline dataprep-webapp/src/app/services/lookup/lookup-service.js]: This does not seem to enough?

**Pull request merged**